src/exec/ticket.rs

```diff
@@ -6,47 +6,39 @@
 // notice may not be copied, modified, or distributed except
 // according to those terms.
 
-use std::env;
 use std::process::Command;
 use std::sync::{Arc, Mutex};
 use std::io;
 
-lazy_static! {
-    /// On non-Windows systems, the `SHELL` environment variable will be used to determine the
-    /// preferred shell of choice for execution. Windows will simply use `cmd`.
-    static ref COMMAND: (String, &'static str) = if cfg!(target_os = "windows") {
-        ("cmd".into(), "/C")
-    } else {
-        (env::var("SHELL").unwrap_or_else(|_| "/bin/sh".into()), "-c")
-    };
-}
-
 /// A state that offers access to executing a generated command.
 ///
-/// The ticket holds a mutable reference to a string that contains the command to be executed.
-/// After execution of the the command via the `then_execute()` method, the string will be
-/// cleared so that a new command can be written to the string in the future.
-pub struct CommandTicket<'a> {
-    command: &'a mut String,
+/// The ticket holds the collection of arguments of a command to be executed.
+pub struct CommandTicket {
+    args: Vec<String>,
```

Comment:
Should probably try to eliminate the `Vec` allocations between commands by making this a `&'a mut Vec<String>`, and then implementing `Drop` for `CommandTicket` to clear the args buffer.

Reply:
I dropped the allocation by removing `CommandTicket` altogether. There was no need for `CommandTicket` anymore, as it only collected the arguments which were then immediately used to build and execute the `Command`. Now the `TokenizedCommand` builds the `Command` with the generated arguments directly and uses `execute_command` (previously `CommandTicket::then_execute`) to execute it.